route/route.go

```diff
@@ -804,21 +805,40 @@ func (r *Router) getSpanStatusCode(status *trace.Status) trace.Status_StatusCode
 	return status.Code
 }
 
-func getSampleRateFromAttributes(attributes map[string]interface{}) (int, error) {
+func getSampleRateFromAttributes(attrs map[string]interface{}) (int, error) {
+	var sampleRateKey string
+	if attrs["sampleRate"] != nil {
+		sampleRateKey = "sampleRate"
+	} else if attrs["SampleRate"] != nil {
+		sampleRateKey = "SampleRate"
+	}
+	if len(sampleRateKey) == 0 || attrs[sampleRateKey] == nil {
+		return defaultSampleRate, nil
+	}
+	var sampleRate int
 	var err error
-	sampleRate := defaultSampleRate
-	if attributes["sampleRate"] != nil {
-		switch attributes["sampleRate"].(type) {
-		case string:
-			sampleRate, err = strconv.Atoi(attributes["sampleRate"].(string))
-		case int:
-			sampleRate = attributes["sampleRate"].(int)
-		default:
-			err = fmt.Errorf("Unrecognised sampleRate datatype - %T", attributes["sampleRate"])
+	switch v := attrs[sampleRateKey].(type) {
+	case string:
+		sampleRate, err = strconv.Atoi(v)
```

Comment:
Do we need to check for this also or does it not matter as much here? `strconv.ParseInt(v, 10, 32)`

Reply:
I don't think so. From the Golang docs:

`Atoi is equivalent to ParseInt(s, 10, 0), converted to type int.`

So I think this is safe enough.

Reply:
> Do we need to check for this also or does it not matter as much here? `strconv.ParseInt(v, 10, 32)`

Yeah, good point - let's keep it consistent with shepherd. I'll update.